examples/bin/run-java

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -eu
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+WHEREAMI="$(dirname "$0")"
+JAVA_BIN="$(source "$WHEREAMI"/java-util && get_java_bin_dir)/java"
+if [ -z "$JAVA_BIN" ]; then
+  >&2 echo "Could not find java - please run $WHEREAMI/verify-java to confirm it is installed."
+  exit 1
+fi
+
+JAVA_MAJOR="$("$JAVA_BIN" -version 2>&1 | grep ' version ' | head -n1 | sed -E 's/.*version "([^"]+)".*/\1/' | cut -d. -f1)"
```

Comment:
you can avoid the `grep ' version ' | head -n1 [...]` by adding `-n` to sed and printing the matching version.
With a little tweaking you can also avoid the `cut -d. -f1`

```suggestion
JAVA_MAJOR="$("$JAVA_BIN" -version 2>&1 | sed -n -E 's/.* version "([^."]*).*/\1/p')"
```

Reply:
Nice. I took your suggestion.